sdks/python/apache_beam/io/avroio.py

```diff
@@ -66,19 +68,22 @@ def __init__(self, label=None, file_pattern=None, min_bundle_size=0,
     Args:
       label: label of the PTransform.
       file_pattern: the set of files to be read.
+      coder: a coder to use to decode values (iff the record type is BYTES).
```

Comment:
I split off the non-coder stuff into https://github.com/apache/incubator-beam/pull/1000 so we can get that in while continuing the discussions on coders here (or perhaps the wider team). I added coders because one may want to store values that don't fit into a schema. 

I'll add a check that if a coder is provided the schema must be bytes. Unfortunately I can't check this until I open the files at runtime. 

In all our sources the coder is used to decode from the external source. (Yes, due to this, produces elements that can be encoded by this Coder too.) It's true that Coders are redundant for any source/sink (e.g. rather than TextIO taking coders, it could always produce and consume Strings, and one would insert DoFns before and after. Here, of course, there's a sane default coder. Likewise for TFRecords, and internally RecordIO, SSTable, etc. There's no need to take coders, but it's quite natural for the serialization format to be specified at the same place as the container format.)